### PR TITLE
chore(email): update styling, copy, urls

### LINKED ISSUE #765: Improve all emails to our users
Unkey sends out a number of emails but we need the following to be rewritten or added.

1. On sign up
2. Day 3 (Welcome from James)
3. Trial ends
4. Subbed
5. Churn

We can now implement resend for the drip emails and don't need to hand spin them for on sign up, and a day 3 email.

### PR DESCRIPTION
**Pull request by mcstepp**
## What does this PR do?

Transactional emails now follow a consistent styling that is legible in light and dark modes with a default template that includes social media connection links. 

Fixes #765 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Run internal/resend locally. Use Resend's "send email" to deliver test emails. 

Key leaked (Desktop, Light)

Screenshot 2024-09-30 at 10 16 33 AM

----------------------------

Payment Issue (Desktop, Light)

Screenshot 2024-09-30 at 10 16 55 AM

 ------------------------------

Payment Issue (Mobile, Dark)
Screenshot_20240930-103520

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Layout` component for improved email structure.
	- Added a `Signature` component for personalized sign-offs.
	- Enhanced email formatting with improved readability and styling across various email notifications.
	- Updated upgrade buttons with new styling and links for better user experience.
	- Reorganized content in emails for clarity and engagement.

- **Bug Fixes**
	- Improved button styles for better visual consistency across emails.

- **Chores**
	- Removed unnecessary dependencies to streamline email components.

### CODE REVIEW

File: internal/resend/emails/payment_issue.tsx [hunk 1]
```diff
@@ -1,62 +1,47 @@
...
-              <Button
-                href="https://unkey.dev/app/settings/billing/stripe"
-                className="px-4 py-2 text-white bg-black rounded"
-              >
-                Update payment information
-              </Button>
-            </Container>
+      <Section className="text-center py-3">
+        <Button
+          href="https://unkey.com/app/settings/billing/stripe"
```
Comment (perkinsjr) [inline internal/resend/emails/payment_issue.tsx:30]: We should change the urls to 

https://app.unkey.com/settings/billing/stripe

File: internal/resend/emails/subscription_ended.tsx [hunk 2]
```diff
@@ -1,64 +1,49 @@
...
-              <Button
-                href="https://unkey.dev/app/settings/billing"
-                className="px-4 py-2 text-white bg-black rounded"
-              >
-                Upgrade Now
-              </Button>
-            </Container>
+      <Section className="text-center py-3">
+        <Button
+          href="https://unkey.com/app/settings/billing"
```
Comment (perkinsjr) [inline internal/resend/emails/subscription_ended.tsx:33]: Same here: 

https://app.unkey.com/settings/billing

File: internal/resend/emails/trial_ended.tsx [hunk 3]
```diff
@@ -1,79 +1,87 @@
...
-              <Button
-                href="https://unkey.dev/app/settings/billing"
-                className="px-4 py-2 text-white bg-black rounded"
-              >
-                Upgrade Now
-              </Button>
-            </Container>
+      <Section className="text-center py-3">
+        <Button
+          href="https://unkey.com/app/settings/billing"
```
Comment (perkinsjr) [inline internal/resend/emails/trial_ended.tsx:69]: Same here : https://app.unkey.com/settings/billing

File: internal/resend/emails/welcome_email.tsx [hunk 4]
```diff
@@ -1,73 +1,71 @@
...
+          <li className="pt-4">
+            {" "}
+            <Link href="https://unkey.com/discord">Unkey Community Discord </Link>
+          </li>
+        </ul>
+      </Section>
+
+      <Section className="text-center py-3">
+        <Button
+          href="https://unkey.com/app"
```
Comment (perkinsjr) [inline internal/resend/emails/welcome_email.tsx:53]: Change this: https://app.unkey.com

**Review by perkinsjr — CHANGES_REQUESTED** [review]

**Comment by perkinsjr** [comment]
✏️ [Changes requested](https://github.com/unkeyed/unkey/pull/2141#pullrequestreview-2337836842)

**Comment by mcstepp** [comment]
Ahh yeah, I thought so but then I saw the original urla in the emails and
second-guessed

File: internal/resend/emails/welcome_email.tsx [hunk 5]
```diff
@@ -1,73 +1,71 @@
...
-            <Text className="text-xs">
-              P.S. - if you have any questions or feedback, reply to this email. I read and reply to
-              every single one.
-            </Text>
-          </Container>
-        </Section>
-      </Html>
-    </Tailwind>
+    <Layout>
+      <Heading className="font-sans text-3xl text-semibold text-center">Welcome to Unkey!</Heading>
```
[the remark this answers is not in the document]
Comment (mcstepp) [reply internal/resend/emails/welcome_email.tsx]: fixed

File: internal/resend/emails/payment_issue.tsx (hunk 1 above)
Comment (mcstepp) [reply internal/resend/emails/payment_issue.tsx]: 👍

**Comment by perkinsjr** [comment]
Hey @mcstepp can you verify why the test is failing for resend

```tsx
ReferenceError: React is not defined
 ❯ Layout src/components/layout.tsx:17:3
     15| 
     16| export const Layout: React.FC<LayoutProps> = ({ children }) => (
     17|   <Html>
       |   ^
     18|     <Tailwind>
     19|       <Head />
```

**Review by perkinsjr — CHANGES_REQUESTED** [review]
See comment about failed tests

**Pull request merged**